### PR TITLE
[Lesson 17_Homework 17] :

### PR DESCRIPTION
**Pull request by TatyanaDovzhich**
- Create method “Navigate to Profile Page via header My Profile button”
- Added pages/profile_page.py & constants/profile_pages.py
- Added tests for profile page (verify Username, Sign Out via Sign In) with related constants and pages
- Added test for hello page (Sign Out via Sign In) with related constants and pages

### CODE REVIEW

File: tests/test_profile_page.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        header.my_profile_button()
+
+        # Navigate to profile page
+        profile_page = hello_page.navigate_to_profile_page()
+        return profile_page
+
+        # Verify profile username
+        profile_page.verify_profile_page_message(username=random_login.username_value)
+
+    def sign_out_from_profile_page_after_sign_in(self, start_page, random_login):
```
Comment (QALightLecorPython) [inline tests/test_profile_page.py]: Test name should start with word 'test_'

File: tests/test_profile_page.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        # Verify login success
+        hello_page.verify_sign_up_message(username=random_login.username_value)
+
+        # Click on "My Profile" button
+        header.my_profile_button()
+
+        # Navigate to profile page
+        profile_page = hello_page.navigate_to_profile_page()
+        return profile_page
```
Comment (QALightLecorPython) [inline tests/test_profile_page.py]: 1. You cannot do return from test
2. You cannot return twice

File: tests/test_profile_page.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        # Verify profile username
+        profile_page.verify_profile_page_message(username=random_login.username_value)
```
Comment (QALightLecorPython) [inline tests/test_profile_page.py:35]: Why do you do verifications under the fixture ?

File: tests/test_hello_page.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            - Navigate to start page
+        """
+        # Sign In as a user
+        hello_page = start_page.sign_in(random_login)
+
+        # Verify login success
+        hello_page.verify_sign_up_message(username=random_login.username_value)
+
+        # Click on "Sign Out" button
+        header.sign_out_button()
```
Comment (QALightLecorPython) [inline tests/test_hello_page.py]: What is header ? From where did you get it ?

File: tests/test_hello_page.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+from constants import header
+
+
+class TestHelloPage:
+    """Stores tests for create hello page base functionality"""
+
+    log = logging.getLogger("[TestHelloPage]")
+
+    def sign_out_from_hello_page_after_sign_in(self, start_page, random_login):
```
Comment (QALightLecorPython) [inline tests/test_hello_page.py]: Is this test ? If so it should starts with 'test_'

File: pages/header.py [hunk 6]
```diff
@@ -36,16 +39,37 @@ def sign_out_button(self, username, password):
...
+        return StartPage(self.driver)
+
+    @wait_until_ok(timeout=3, period=0.5)
+    def click_and_validate_my_profile_button(self):
+        """Click on My Profile button until it disappear"""
+        self.click(self.const.MY_PROFILE_BUTTON_XPATH)
+        assert not self.is_element_exists(self.const.MY_PROFILE_BUTTON_XPATH), "My Profile button didn't disappear"
+
+    @log_wrapper
+    def my_profile_button(self, username, password):
```
Comment (QALightLecorPython) [inline pages/header.py:56]: Do you really need such validation for this button ?

**Review by QALightLecorPython — CHANGES_REQUESTED** [review]
Looks good, but I have a few comments

File: tests/test_profile_page.py (hunk 1 above)
Comment (TatyanaDovzhich) [reply tests/test_profile_page.py]: Fixed: test  was renamed into "test_navigate_to_profile_page_via_sign_in"

File: tests/test_profile_page.py (hunk 3 above)
Comment (TatyanaDovzhich) [reply tests/test_profile_page.py:35]: image
I check the username message on the page https://qa-complexapp.onrender.com/profile/

File: tests/test_hello_page.py (hunk 5 above)
Comment (TatyanaDovzhich) [reply tests/test_hello_page.py]: Fixed:  test  was renamed into "test_sign_out_from_profile_page_after_sign_in"

File: pages/header.py (hunk 6 above)
Comment (TatyanaDovzhich) [reply pages/header.py:56]: Fixed: validation method for My Profile button was deleted

File: tests/test_hello_page.py (hunk 4 above)
Comment (TatyanaDovzhich) [reply tests/test_hello_page.py]: Fixed: reference to hello_page was added into Click on "Sign Out" button & Click on "My Profile" button Steps (tests/test_hello_page.py & tests/test_profile_page.py

File: tests/test_profile_page.py (hunk 2 above)
Comment (TatyanaDovzhich) [reply tests/test_profile_page.py]: Fixed: strings 35 and 61 "return profile_page" was deleted

**Pull request merged**